README.md

```diff
@@ -0,0 +1,568 @@
+
+
+### 6.3 Result Score
+ViSearch image search results are ranked in descending order i.e. from the highest scores to the lowest, ranging from 1.0 to 0.0. By default, the score for each result is not returned. You can turn on the score parameter to retrieve the scores for each image result:
+
+```Swift
+...
+
+params.score = true; // result will include score for every image
+
+// start searching
+...
+
+```
+
+If you need to restrict search results from a minimum score to a maximum score, specify the score_min and/or score_max parameters:
+```swift
+...
+
+params.score = true; // result will include score for every image
+params.scoreMin = 0.3; // the minimum score is 0.3  
+params.scoreMax = 0.8; // the maximum score is 0.8
+
+// start searching. Every image result will have a score within [0.3, 0.8].
+...
+```
+
+### 6.4 Automatic Object Recognition Beta
+With Automatic Object Recognition, ViSearch /uploadsearch API is smart to detect the objects present in the query image and suggest the best matched product type to run the search on.
+
+You can turn on the feature in upload search by setting the API parameter "detection=all". We are now able to detect various types of fashion items, including `Top`, `Dress`, `Bottom`, `Shoe`, `Bag`, `Watch` and `Indian Ethnic Wear`. The list is ever-expanding as we explore this feature for other categories.
+
+Notice: This feature is currently available for fashion application type only. You will need to make sure your app type is configurated as "fashion" on [ViSenze dashboard](https://developers.visenze.com/setup/#Choose-Your-Application-Type).
+
+```swift
+params.detection = "all";
+```
+You can use the Box parameter to restrict the image area [x1, y1, x2, y2] as the portion of your image to search for. When you input a box with 0 width and 0 height, eg. “box”:[574,224,574,224]. We will treat it as a point and detect the object over the current point.
+
+![](https://developers.visenze.com/api/images/detection_point.png)
+
+You could also recognize objects from a paticular type on the uploaded query image through configuring the detection parameter to a specific product type as "detection={type}". Our API will run the search within that product type.
+
+Sample request to detect `bag` in an uploaded image:
+
+```swift
+params.detection = "bag";
+```
+
+The detected product types are listed in `product_types` together with the match score and box area of the detected object. Multiple objects can be detected from the query image and they are ranked from the highest score to lowest. The full list of supported product types by our API will also be returned in `product_types_list`.
+
+## 7. Event Tracking
+
+### Send Action For Tracking
+User action (e.g. click on an image after search) can be sent in this way:
+
+```swift
+
+let params = ViTrackParams(accessKey: ViSearch.sharedInstance.client!.accessKey, reqId: recentReqId, action: "click" )
```

Comment:
That is because the ViTrackParams is an independent class used to create the tracking request which cannot get the access key directly from the client. To make it consistent with other API methods ( recommendation, findSimilar, etc in ViSearchClient) that take in sub-classes of ViBaseSearchParams type, it was designed that way.  I will make the necessary changes to the constructor to make accessKey optional. Will update Readme and code after the change.

```
 @discardableResult public func uploadSearch(params: ViUploadSearchParams,
                             successHandler: @escaping SuccessHandler,
                             failureHandler: @escaping FailureHandler) -> URLSessionTask
   
 @discardableResult public func colorSearch(params: ViColorSearchParams,
                            successHandler: @escaping SuccessHandler,
                            failureHandler: @escaping FailureHandler
                            ) -> URLSessionTask
 
...
```